test/nodetool/conftest.py

```diff
@@ -27,31 +27,52 @@ def pytest_addoption(parser):
                      " with --nodetool=cassandra, this should be the nodetool binary")
     parser.addoption('--jmx-path', action='store', default=None,
                      help="Path to the jmx binary, only used with --nodetool=cassandra")
+    parser.addoption('--run-within-unshare', action='store_true',
+                     help="Setup the 'lo' network if launched with unshare(1)")
 
 
 @pytest.fixture(scope="session")
-def rest_api_mock_server():
-    ip = f"127.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}"
-    port = random.randint(10000, 65535)
+def maybe_setup_loopback_network(request):
+    # unshare(1) -rn drops us in a new network namespace in which the "lo" is
+    # not up yet, so let's set it up first.
+    if request.config.getoption('--run-within-unshare'):
+        try:
+            args = "ip link set lo up".split()
+            subprocess.run(args, check=True)
+        except FileNotFoundError:
+            args = "/sbin/ifconfig lo up".split()
+            subprocess.run(args, check=True)
+    yield
```

Comment:
Not required, when there is no state/value to be kept alive until teardown.

Reply:
right. but it's just a way to structure the code in order to make this more explicit.